### DISCUSSION TITLE
Feature Request: Allow multi-datasource queries

### DISCUSSION THREAD

**Discussion by gotjosh** [grafana/grafana#48729] [Alerting]
Now that #39249 is possible within unified alerting, we can have multiple alerts that spawn from a single query. This is equivalent to having template variables in an alert as you'd get one single alert per combination of labels that the query outputs.

Unless... One of your template variables is the datasource. 

Alert Queries are defined at a datasource-level, meaning that while technically possible to define multiple queries that have different datasources - in practice, this is not equal.

First, we don't have a merge function as part of expressions. Assuming: 

Query A: `sum (up) by (cluster, namespace)` and `datasource=prometheus-a`
Query B: `sum (up) by (cluster, namespace)` and `datasource=prometheus-b`
Expression: `$A + B or $A || $B` **does not work**

This is because if labels are equal from both datasources, there's no way to differentiate if a series comes from one datasource or another.

Second, it should be much more intuitive and clear than what we have defined above.

A query and merge operation from multiple datasources should be very clear on what the outcomes is e.g. we'll add a synthetic label with the name of the datasource to each series that's coming from that specific datasource.

I'm not sure how common it is to have the same alert for multiple datasources - all this is allivating is the fact that you won't have to create an alert per datasource.

Opening this as a feature request to gather some input from the community - drop your comments below 👌

**Comment by Himura2la** [comment]
Yes, this must be the issue! I searched across several threads and found no soluton

**Comment by armandgrillet** [comment]
Closing in favor of https://github.com/grafana/grafana/issues/73920 as feature requests should now be issues.
